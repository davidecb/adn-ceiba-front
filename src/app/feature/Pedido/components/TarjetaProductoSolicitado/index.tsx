import * as PropTypes from 'prop-types';
import * as React from 'react';
import {
  Acabados,
  CaracteristicasProducto,
  CostoProducto,
  Eliminar,
  NombreProducto,
  ProductoImagen,
  TarjetaProductoSolicitadoContainer,
  TiempoEjecucion
} from './styles';
import { ProductoSolicitado } from '../../../Producto/models/ProductoSolicitado';


export interface TarjetaProductoSolicitadoProps {
  productoSolicitado: any;
  onClickPersonalizarProducto: (productoSolicitado: ProductoSolicitado) => void;
  eliminarProductoSolicitado: (idProductoSolicitado: number) => void;
}

export const TarjetaProductoSolicitado: React.FC<TarjetaProductoSolicitadoProps> = ({
  productoSolicitado,
  onClickPersonalizarProducto,
  eliminarProductoSolicitado,
}) => {
  const handleEliminar = () => {
    eliminarProductoSolicitado(productoSolicitado.id);
    window.location.reload();
  };

  return (
    <TarjetaProductoSolicitadoContainer>
      <Eliminar onClick={handleEliminar}>‚ùå</Eliminar>
      <ProductoImagen src={productoSolicitado.productoSolicitado.producto.imagen} />
      <NombreProducto>{productoSolicitado.productoSolicitado.producto.nombre}</NombreProducto>
      <CostoProducto><b>costo/unidad:</b> ${productoSolicitado.productoSolicitado.costo}</CostoProducto>
      <TiempoEjecucion><b>tiempo/unidad:</b> {productoSolicitado.productoSolicitado.tiempo} min.</TiempoEjecucion>
      <CaracteristicasProducto>
        <div><b>material:</b> {productoSolicitado.productoSolicitado.material}</div>
        <div><b>color:</b> {productoSolicitado.productoSolicitado.color}</div>
        <div>
          <b>acabado:</b>
          <Acabados>
            <div>pulido: <input type='checkbox' defaultChecked={productoSolicitado.productoSolicitado.acabado.pulido as boolean} /></div>
            <div>pintado: <input type='checkbox' defaultChecked={productoSolicitado.productoSolicitado.acabado.pintado as boolean} /></div>
            <div>barnizado: <input type='checkbox' defaultChecked={productoSolicitado.productoSolicitado.acabado.barnizado as boolean} /></div>
          </Acabados>
        </div>
        <div>
          <b>urgencia:</b> 
          <input type='checkbox' defaultChecked={productoSolicitado.productoSolicitado.urgencia as boolean} />
        </div>
        <div><b>cantidad:</b> {productoSolicitado.cantidad}</div>
      </CaracteristicasProducto>
      <CostoProducto><b>costo total:</b> ${productoSolicitado.productoSolicitado.costo * productoSolicitado.cantidad}</CostoProducto>
      <TiempoEjecucion><b>tiempo total:</b> {productoSolicitado.productoSolicitado.tiempo * productoSolicitado.cantidad} min.</TiempoEjecucion>
    </TarjetaProductoSolicitadoContainer>
  );
};

TarjetaProductoSolicitado.propTypes = {
  productoSolicitado: PropTypes.any.isRequired,
  onClickPersonalizarProducto: PropTypes.func.isRequired,
  eliminarProductoSolicitado: PropTypes.func.isRequired,
};
