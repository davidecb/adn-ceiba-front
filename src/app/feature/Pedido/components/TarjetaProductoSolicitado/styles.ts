import styled from 'styled-components';

export const TarjetaProductoSolicitadoContainer = styled.div`
  width: 250px;
  margin: 10px;
  padding: 10px;
  border: 2px groove rgb(10, 10, 114);
  border-radius: 10px;
  display: flex;
  flex-wrap: wrap;
  flex-flow: column;
  align-items: center;
  justify-content: center;
`;

export const Eliminar = styled.div`
width: 100%;
text-align: rigth;
cursor: pointer;
`;

export const NombreProducto = styled.div`
  width: 100%;
  font-size: large;
  text-align: center;
  font-weight: bolder;
`;

export const CostoProducto = styled.span`
  margin: 0 10px;
  font-size: large;
  text-align: center;
`;

export const TiempoEjecucion = styled.span`
  margin: 0 10px;
  font-size: large;
  text-align: center;
`;

export const Acabados = styled.div`
width: 100%;
display: flex;
flex-direction: column;
margin-left: 20px;  
`;

export const CaracteristicasProducto = styled.div`
  width: 100%;
  padding: 5px;
  font-size: medium;
  border: 2px groove rgb(200, 200, 200);
  border-radius: 10px;
  pointer-events: none;
`;

export const ProductoImagen = styled.img`
  border: 3px outset rgb(200, 200, 200);
  border-radius: 10px;
  min-height: 180px;
  min-width: 200px;
  max-height: 180px;
  max-width: 200px;  
  object-fit: contain;
`;
