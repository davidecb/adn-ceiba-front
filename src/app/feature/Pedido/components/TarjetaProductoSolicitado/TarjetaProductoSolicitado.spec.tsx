import * as React from 'react';
import { RenderResult, render } from '@testing-library/react';
import { SinonStub, stub } from 'sinon';
import { TarjetaProductoSolicitado } from './../TarjetaProductoSolicitado';

describe('TarjetaProductoSolicitado Test', () => {
  let componentWrapper: RenderResult;
  let componentProps: React.ComponentProps<typeof TarjetaProductoSolicitado> & {
    onClickPersonalizarProducto: SinonStub;
    eliminarProductoSolicitado: SinonStub;
  };

  beforeEach(() => {
    componentProps = {
      productoSolicitado: {
        id: 1,
        pedido: 1,
        cantidad: 2,
        productoSolicitado: {
          material: 'PLA',
          color: 'negro',
          acabado: {
            pulido: false,
            pintado: false,
            barnizado: false
          },
          urgencia: false,
          producto: {
            id: 1,
            nombre: 'producto testing',
            costo: 10000,
            tiempo: 30,
            imagen: 'imagen.jpg'
          }
        }
      },
      onClickPersonalizarProducto: stub(),
      eliminarProductoSolicitado: stub(),
    };
    componentWrapper = render(<TarjetaProductoSolicitado {...componentProps} />);
  });

  it('should match snapshot', () => {
    expect(componentWrapper.container).toMatchSnapshot();
  });
});

export {};
