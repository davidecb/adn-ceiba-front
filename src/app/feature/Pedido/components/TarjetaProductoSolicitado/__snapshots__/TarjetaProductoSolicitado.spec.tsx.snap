// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TarjetaProductoSolicitado Test should match snapshot 1`] = `
<div>
  <div
    class="sc-AxjAm cVcuKC"
  >
    <div
      class="sc-AxirZ elnmyV"
    >
      ‚ùå
    </div>
    <img
      class="sc-Axmtr kBGTeh"
      src="imagen.jpg"
    />
    <div
      class="sc-AxiKw egosaa"
    >
      producto testing
    </div>
    <span
      class="sc-AxhCb cZiLQ"
    >
      <b>
        costo/unidad:
      </b>
       $
    </span>
    <span
      class="sc-AxhUy fvcdQl"
    >
      <b>
        tiempo/unidad:
      </b>
       
       min.
    </span>
    <div
      class="sc-AxheI dZIHTm"
    >
      <div>
        <b>
          material:
        </b>
         
        PLA
      </div>
      <div>
        <b>
          color:
        </b>
         
        negro
      </div>
      <div>
        <b>
          acabado:
        </b>
        <div
          class="sc-AxgMl ipXOHv"
        >
          <div>
            pulido: 
            <input
              type="checkbox"
            />
          </div>
          <div>
            pintado: 
            <input
              type="checkbox"
            />
          </div>
          <div>
            barnizado: 
            <input
              type="checkbox"
            />
          </div>
        </div>
      </div>
      <div>
        <b>
          urgencia:
        </b>
        <input
          type="checkbox"
        />
      </div>
      <div>
        <b>
          cantidad:
        </b>
         
        2
      </div>
    </div>
    <span
      class="sc-AxhCb cZiLQ"
    >
      <b>
        costo total:
      </b>
       $
      NaN
    </span>
    <span
      class="sc-AxhUy fvcdQl"
    >
      <b>
        tiempo total:
      </b>
       
      NaN
       min.
    </span>
  </div>
</div>
`;
